examples/context/theme-detailed-app.js

```diff
@@ -0,0 +1,37 @@
+import ThemeContext from './theme-context';
+import ThemedButton from './button';
+
+class App extends React.Component {
+  state = {
+    theme: {
+      highlight: 'blue',
+      accent: 'purple',
```

Comment:
nit: It's also a little weird that we have a default value and `state` with the same value. People might not understand why default value exists if we always do this.

Reply:
What's a good application use case for the default value? The RFC example just says ["to ensure type correctness"](https://github.com/reactjs/rfcs/blob/master/text/0002-new-version-of-context.md#detailed-design).

Reply:
That's a good observation. I wasn't considering the type-correctness angle. 👍

I was thinking it would be nice for the docs to highlight, in one of the examples, that the default value gets passed to a Consumer if there is no Provider ancestor.